agent/requestor/src/main.rs

```diff
@@ -156,19 +156,15 @@ async fn main() -> anyhow::Result<()> {
     let subscription_id = api.market.subscribe(&my_demand).await?;
     log::info!("\n\n DEMAND SUBSCRIBED: {}", subscription_id);
 
-    let shutdown = shutdown_handler(
-        activities.clone(),
-        agreement_allocation.clone(),
-        subscription_id.clone(),
-        api.clone(),
-    );
+    let (future_ctrl_c, ctrl_c_abort) = futures::future::abortable(signal::ctrl_c());
```

Comment:
I'd rather hide the details of how we process ctrl_c. Can you pls move it back to `shutdown_handler` which in turn will return future to wait for at the end together with abort handler?

Reply:
eg like this
`let (shutdown, app_abort_handle) = shutdown_handler(...)`
then at the end of `main` you can : 
`shutdown.await`
and in the payments:
`app_abort_handle.abort()`